### PR TITLE
Small strict null checking pass on a few files

### PR DESCRIPTION
**Pull request by anthonydresser**
I did a small pass for strict null checking to see what would be involved. These were the changes to get a few files to pass the strict null pass.

### CODE REVIEW

File: src/sql/base/browser/ui/breadcrumb/breadcrumb.component.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ import { IDisposable } from 'vs/base/common/lifecycle';
 				`
 })
 export class BreadcrumbComponent implements OnInit, OnDestroy {
-	private menuItems: MenuItem[] = [];
+	protected menuItems: MenuItem[] = [];
```
Comment (MattIrv) [inline src/sql/base/browser/ui/breadcrumb/breadcrumb.component.ts]: why is this necessary?

File: src/sql/base/browser/ui/button/button.ts [hunk 2]
```diff
@@ -13,11 +13,10 @@ export interface IButtonStyles extends vsIButtonStyles {
 }
 
 export class Button extends vsButton {
-	private buttonFocusOutline: Color;
+	private buttonFocusOutline?: Color;
```
Comment (MattIrv) [inline src/sql/base/browser/ui/button/button.ts]: VS Code seems to prefer using types like `Color | null`. Does it make a difference?

File: src/sql/base/browser/ui/dropdownList/dropdownList.ts [hunk 3]
```diff
@@ -7,18 +7,15 @@
...
 import { Color } from 'vs/base/common/color';
 import { IAction } from 'vs/base/common/actions';
 import { EventType as GestureEventType } from 'vs/base/browser/touch';
 import { List } from 'vs/base/browser/ui/list/listWidget';
 import { StandardKeyboardEvent } from 'vs/base/browser/keyboardEvent';
 import { KeyCode } from 'vs/base/common/keyCodes';
-import { Builder } from 'sql/base/browser/builder';
 
-import { Button } from 'sql/base/browser/ui/button/button';
-import { attachButtonStyler } from 'sql/platform/theme/common/styler';
```
Comment (MattIrv) [inline src/sql/base/browser/ui/dropdownList/dropdownList.ts]: thanks for cleaning these up

File: src/sql/base/browser/ui/editableDropdown/dropdownTree.ts [hunk 4]
```diff
@@ -43,44 +43,44 @@ export class DropdownRenderer implements tree.IRenderer {
...
 	}
 
-	public renderElement(tree: tree.ITree, element: Resource, templateId: string, templateData: Template): void {
+	public renderElement(tree: any, element: Resource, templateId: string, templateData: Template): void {
 		templateData.label.innerText = element.value;
 		templateData.row.title = element.value;
 	}
 
-	public disposeTemplate(tree: tree.ITree, templateId: string, templateData: Template): void {
+	public disposeTemplate(tree: any, templateId: string, templateData: Template): void {
```
Comment (MattIrv) [inline src/sql/base/browser/ui/editableDropdown/dropdownTree.ts]: why are all of these becoming `any`? Is there a more precise type we can use? `any` makes it hard to maintain

File: src/tsconfig.strictNullChecks.json [hunk 5]
```diff
@@ -6,734 +6,746 @@
...
-		"./vs/workbench/services/textMate/electron-browser/textMateService.ts",
-		"./vs/workbench/services/textfile/electron-browser/textResourcePropertiesService.ts",
-		"./vs/workbench/services/themes/common/colorExtensionPoint.ts",
-		"./vs/workbench/services/themes/common/colorThemeSchema.ts",
-		"./vs/workbench/services/themes/common/fileIconThemeSchema.ts",
-		"./vs/workbench/services/themes/common/workbenchThemeService.ts",
-		"./vs/workbench/services/title/common/titleService.ts",
-		"./vs/workbench/services/workspace/common/workspaceEditing.ts",
-		"./vs/workbench/test/electron-browser/api/mock.ts"
+		// "./vs/base/browser/browser.ts",
```
Comment (MattIrv) [inline src/tsconfig.strictNullChecks.json]: Do we need to disable all the VS Code ones? Shouldn't they pass correctly unless we've made changes that don't follow strict null, in which case we should update those changes?

Comment (anthonydresser) [reply src/tsconfig.strictNullChecks.json]: We don't have the work they have done to make all their stuff pass; so including theirs adds a lot of noisy errors. Maybe once we get the next merge most of their errors should be fixed

File: src/sql/base/browser/ui/editableDropdown/dropdownTree.ts (hunk 4 above)
Comment (anthonydresser) [reply src/sql/base/browser/ui/editableDropdown/dropdownTree.ts]: Its mainly since those aren't being used by the function but are necessary parts of the function signiture. I could make them `tree.ITree | undefined` to be cleaner

File: src/sql/base/browser/ui/breadcrumb/breadcrumb.component.ts (hunk 1 above)
Comment (anthonydresser) [reply src/sql/base/browser/ui/breadcrumb/breadcrumb.component.ts]: its the cleaniest way to defined a variable that isn't being used by the class directly but is being used in the html template. Since the tslinter can't parse angular code it will report this as being unused and throw a type error, even if it is being used in the template. The other solution is to type ignore it (which is a valid approach) or make it public (which I don't think is the right approach)

File: src/sql/base/browser/ui/button/button.ts (hunk 2 above)
Comment (anthonydresser) [reply src/sql/base/browser/ui/button/button.ts]: They do it differently in different areas. HTMLElement style properties are defined as `type | null` so you can't assign them to be `undefined` (`undefined != null`). But most places we use undefined, and the `?` specifier to translated as `undefined` not `null` so in places where you want to directly assign a variable to a html property it can be cleaner to use `Color | null` otherwise it's cleaner to use `?: Color`.

Comment (anthonydresser) [reply src/sql/base/browser/ui/button/button.ts]: Specifically for `color` since you never set a html property to it directly, its probably always correct to use `?: Color`. But I think with vscode it mainly comes down to who wrote it.
https://github.com/Microsoft/azuredatastudio/blob/master/src/vs/editor/contrib/gotoError/gotoErrorWidget.ts#L167
https://github.com/Microsoft/azuredatastudio/blob/master/src/vs/base/browser/ui/menu/menu.ts#L36
or even
`?: Color | null`
https://github.com/Microsoft/azuredatastudio/blob/master/src/vs/editor/contrib/zoneWidget/zoneWidget.ts#L34

Comment (anthonydresser) [reply src/sql/base/browser/ui/button/button.ts]: From a javascript point of view `undefined` means the value was never set where as `null` is meaning you set the value to not exist. But it's really just semantics.
https://www.ajaymatharu.com/javascript-difference-between-undefined-and-null/

File: src/sql/base/browser/ui/editableDropdown/dropdownTree.ts (hunk 4 above)
Comment (MattIrv) [reply src/sql/base/browser/ui/editableDropdown/dropdownTree.ts]: That’s probably better just in case anyone comes along later who wants to use them

File: src/sql/base/browser/ui/breadcrumb/breadcrumb.component.ts (hunk 1 above)
Comment (anthonydresser) [reply src/sql/base/browser/ui/breadcrumb/breadcrumb.component.ts]: More I think about it, the more I think it might be cleaner here to just add a tslint disable tag and add a comment that says that the variable is used in the angular template.

Comment (anthonydresser) [reply src/sql/base/browser/ui/breadcrumb/breadcrumb.component.ts]: nvm this can't be fixed via tslint rules since this is a compiler feature not a tslint feature
